dump.go

```diff
@@ -84,8 +84,13 @@ func (d *dumpRoundTripper) RoundTrip(request *http.Request) (response *http.Resp
 		if err != nil {
 			return
 		}
-		d.dumpResponse(ctx, response, body)
-		response.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		if len(body) == 0 {
+			d.dumpResponse(ctx, response, nil)
+			response.Body = http.NoBody // checked by Unmarshal* functions.
```

Comment:
Specifically, this check:
https://github.com/openshift-online/ocm-sdk-go/blob/573f167175c9144be490617bd85acc40458e216f/clustersmgmt/v1/cluster_type_json.go#L413-L416

UnmarshalCluster already wanted to return nil for empty bodies but was being fooled by the replacement `ioutil.NopCloser(bytes.NewBuffer(body))` from this middleware — this is why it only crashed in debug mode.